### PR TITLE
routers/api: add constraints to API fields

### PR DESCRIPTION
**Pull request by efahl**
Use pydantic's Field to add constraints to 'defaults' (string length <= 20480) and 'rootfs_size_mb' (int 1 <= value <= 1024) 'package_changes'

Reference:
https://docs.pydantic.dev/latest/concepts/fields/#numeric-constraints

Signed-off-by: Eric Fahlgren @efahl

### CODE REVIEW

File: asu/routers/api.py [hunk 1]
```diff
@@ -20,9 +20,21 @@ class BuildRequest(BaseModel):
     packages: Union[list, None] = []
     profile: str
     packages_versions: dict = {}
-    defaults: Union[str, None] = None
+    defaults: Optional[
+        Annotated[
+            str,
+            Field(default=None, max_length=20480, description="asu-defaults script"),
```
Comment (aparcar) [inline asu/routers/api.py:26]: It's called uci-defaults, just to avoid confusion :)

File: asu/routers/api.py [hunk 2]
```diff
@@ -20,9 +20,21 @@ class BuildRequest(BaseModel):
...
+        ]
+    ] = None
     client: Union[str, None] = "unknown/0"
-    rootfs_size_mb: Union[int, None] = None
+    rootfs_size_mb: Optional[
+        Annotated[
+            int,
+            Field(
+                default=None, gt=0, le=1024, description="Resize the root file system."
```
Comment (aparcar) [inline asu/routers/api.py:34]: Could you add this to the config.py please to make it configurable?

File: asu/routers/api.py (hunk 1 above)
Comment (aparcar) [inline asu/routers/api.py:26]: Could you add this to the config.py please to make it configurable?

**Comment by aparcar** [comment]
Thank for this PR, looks great!

**Comment by efahl** [comment]
While I'm poking around here...

What do you think about changing the other `Union[type, None]` declarations into `Optional[type]`? (At least for the ones right around the ones I'm changing.)  As I recall, `Optional` was added to typing way back in 3.8, and it seems to be the more common idiom these days.

File: asu/routers/api.py (hunk 1 above)
Comment (efahl) [reply asu/routers/api.py:26]: I will go back to old openapi.yml and steal the comments from there.  Not sure why I didn't think of that sooner...

**Comment by aparcar** [comment]
Please go ahead, this typing stuff is rather new for me

File: asu/routers/api.py (hunk 1 above)
Comment (efahl) [reply asu/routers/api.py]: Put the yaml comments into the new code.

File: asu/routers/api.py (hunk 2 above)
Comment (efahl) [reply asu/routers/api.py]: All constants removed and now come from `settings...`, including the description for `defaults`.

**Comment by aparcar** [comment]
Excellent thank you very much. Will deploy this version tomorrow

**Pull request merged**